wicketd/src/lib.rs

```diff
@@ -53,7 +57,50 @@ pub struct Args {
     pub address: SocketAddrV6,
     pub artifact_address: SocketAddrV6,
     pub mgs_address: SocketAddrV6,
+    pub nexus_proxy_address: SocketAddrV6,
     pub baseboard: Option<Baseboard>,
+    pub rack_subnet: Option<Ipv6Subnet<AZ_PREFIX>>,
+}
+
+pub struct SmfConfigValues {
+    pub rack_subnet: Option<Ipv6Subnet<AZ_PREFIX>>,
+}
+
+impl SmfConfigValues {
+    #[cfg(target_os = "illumos")]
+    pub fn read_current() -> Result<Option<Self>> {
+        use anyhow::Context;
+        use illumos_utils::scf::ScfHandle;
+
+        const CONFIG_PG: &str = "config";
+        const PROP_RACK_SUBNET: &str = "rack-subnet";
+
+        let scf = ScfHandle::new()?;
+        let instance = scf.self_instance()?;
+        let snapshot = instance.running_snapshot()?;
+        let config = snapshot.property_group(CONFIG_PG)?;
+
+        let rack_subnet = config.value_as_string(PROP_RACK_SUBNET)?;
+
+        let rack_subnet = if rack_subnet == "unknown" {
+            None
+        } else {
+            let addr = rack_subnet.parse().with_context(|| {
+                format!(
+                    "failed to parse {CONFIG_PG}/{PROP_RACK_SUBNET} \
+                     value {rack_subnet:?} as an IP address"
+                )
+            })?;
+            Some(Ipv6Subnet::new(addr))
+        };
+
+        Ok(Some(Self { rack_subnet }))
+    }
+
+    #[cfg(not(target_os = "illumos"))]
+    pub fn read_current() -> Result<Option<Self>> {
+        Ok(None)
```

Comment:
Wouldn't it be better to return an error here? Is there a case where that breaks down?

Reply:
It would be better! This choice went along with the way the `wicketd` clap arguments were being parsed in an earlier version, and I didn't realize I could clean this up at this point. Changed in [53e0654](https://github.com/oxidecomputer/omicron/pull/4224/commits/53e06542b7c3b57722a12e672c35d49d3afb3d17)